academic-services/knowledge-exploration-service/reference-get-histogram.md

```diff
@@ -0,0 +1,223 @@
+`attributes` | Optional | string | A list of comma-separated attributes to generate histograms for. See the [entity schema](reference-makes-api-entity-schema.md) for the attributes that can be requested. If no attributes are specified the response will still include the total number of matching entities. <br/><br/>Defaults to an empty string.
+`offset` | Optional | integer | The number of histogram values to skip. <br/><br/>Defaults to 0.
+`count` | Optional | integer | The number of histogram values to return for each attribute. <br/><br/>Defaults to 10.
+
+## Responses
+
+Name | Type | Description
+--- | --- | ---
+200 OK | [HistogramResponse](#histogramresponse) | Histogram response was successfully generated and returned.
+
+## Definitions
+
+| | |
+| --- | --- |
+[HistogramAttribute](#histogramattribute) | Information for an individual attribute
+[HistogramAttributeValue](#histogramattributevalue) | Information for a distinct attribute value
+[HistogramResponse](#histogramresponse) | Histogram response information
+
+### HistogramAttribute
+
+Name | Type | Description
+--- | --- | ---
+attribute | string | The name of the attribute which the histogram was computed from
+total_count | long | Total sum of HistogramAttributeValue.count for each attribute
+histogram | [HistogramAttributeValue](#histogramattributevalue)[] | Histogram value distribution for this attribute
+
+### HistogramAttributeValue
+
+Name | Type | Description
+--- | --- | ---
+count | integer | Number of entities that contain this attribute value
+logprob | double | Total log probability of entities with this attribute value
+value | string | Attribute value
+
+### HistogramResponse
+
+Name | Type | Description
+--- | --- | ---
+expr | string | The query expression used when generating this response
+histograms | [HistogramAttribute](#histogramattribute)[] | An array of containing each requested attribute histogram
+num_entities | integer | The number of matching entities that were used to generate histograms
+
+## Examples
+
+### Retrieve top fields of study and conferences that Microsoft published papers in during 2019
+
+#### Request headers
+
+```http
+GET /calchistogram?expr=And(Composite(AA.AfN=%27microsoft%27),Y=2019)&attributes=C.CN,F.DFN&offset=0&count=10 HTTP/1.1
+Host: makesexample.westus.cloudapp.azure.com
+Connection: keep-alive
+Upgrade-Insecure-Requests: 1
+User-Agent: contoso/1.0
+Accept: application/json
+Accept-Encoding: gzip, deflate
+Accept-Language: en-US,en;q=0.9
+```
+
+#### Response header
+
+```http
+HTTP/1.1 200 OK
+Transfer-Encoding: chunked
+Content-Type: application/json; charset=utf-8
+Server: Kestrel
+X-Powered-By: ASP.NET
+Date: Thu, 30 Jan 2020 01:13:55 GMT
+```
+
+#### Response payload
+
+```json
+{
+    "expr": "And(Composite(AA.AfN='microsoft'),Y=2019)",
+    "num_entities": 4105,
+    "histograms": [
+        {
+            "attribute": "F.DFN",
+            "distinct_values": 29487,
```

Comment:
>distinct_values [](start = 13, length = 15)

no longer supported

Reply:
Have a separate API change I'm working on to remove this, but will remove here as well. Good catch!